plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/files/PropertyReload.java

```diff
@@ -0,0 +1,194 @@
+        if (beanFactoryBeanNamesForTypes != null) {
+            for (String beanFactoryBeanName : beanFactoryBeanNamesForTypes) {
+                PlaceholderConfigurerSupport placeholderConfigurerSupport = beanFactory.getBean(beanFactoryBeanName, PlaceholderConfigurerSupport.class);
+                try {
+                    updatePlaceholderConfigurerSupport(beanFactory, placeholderConfigurerSupport);
+                } catch (Exception e) {
+                }
+            }
+        }
+    }
+
+
+    private static void updatePlaceholderConfigurerSupport(DefaultListableBeanFactory beanFactory, PlaceholderConfigurerSupport placeholderConfigurerSupport) {
+        if (placeholderConfigurerSupport instanceof PropertySourcesPlaceholderConfigurer) {
+            PropertySourcesPlaceholderConfigurer propertySourcesPlaceholderConfigurer = (PropertySourcesPlaceholderConfigurer) placeholderConfigurerSupport;
+            Field field = null;
+            try {
+                field = PropertySourcesPlaceholderConfigurer.class.getDeclaredField("propertySources");
+                field.setAccessible(true);
+                MutablePropertySources origPropertySources = (MutablePropertySources) field.get(propertySourcesPlaceholderConfigurer);
+                origPropertySources.forEach(propertySource -> {
+                    origPropertySources.remove(propertySource.getName());
+                });
+                field.set(propertySourcesPlaceholderConfigurer, null);
+                propertySourcesPlaceholderConfigurer.postProcessBeanFactory(beanFactory);
+
+                MutablePropertySources curPropertySources = (MutablePropertySources) field.get(propertySourcesPlaceholderConfigurer);
+                curPropertySources.forEach(propertySource -> {
+                    origPropertySources.addLast(propertySource);
+                });
+                field.set(propertySourcesPlaceholderConfigurer, origPropertySources);
+            } catch (NoSuchFieldException e) {
+                LOGGER.debug("Failed to reload PropertySourcesPlaceholderConfigurer, possibly Spring version is 3.x or less", e);
+            } catch (IllegalAccessException e) {
+                LOGGER.debug("Failed to reload PropertySourcesPlaceholderConfigurer, possibly Spring version is 3.x or less", e);
+            }
+        } else if (placeholderConfigurerSupport instanceof PropertyPlaceholderConfigurer) {
+            PropertyPlaceholderConfigurer propertyPlaceholderConfigurer = (PropertyPlaceholderConfigurer) placeholderConfigurerSupport;
+            propertyPlaceholderConfigurer.postProcessBeanFactory(beanFactory);
+        }
+    }
+
+    /**
+     * Deal with the condition:
+     * 1. constructor contains @Value parameter
+     * 2. @Bean method contains @Value parameter
+     *
+     * @param beanFactory
+     * @return
+     */
+    public static Set<String> checkAndGetReloadBeanNames(DefaultListableBeanFactory beanFactory) {
```

Comment:
The method name is inappropriate. It looks to me it handles AnnotatedBeanDefinition only.